core/shared/src/main/scala/monocle/function/Plated.scala

```diff
@@ -62,4 +65,19 @@ trait PlatedFunctions {
   def transformOf[A](l: Setter[A, A])(f: A => A)(a: A): A =
     l.modify(b => transformOf(l)(f)(f(b)))(a)
 
-}
+  /** transforming counting changes */
+  def transformC[A: Plated](pf: PartialFunction[A, A])(a: A): (Int, A) = {
+    transformM[A, State[Int, ?]](
+      pf.andThen(b => State((i: Int) => (i + 1, b)))
+        .applyOrElse(_, State.state))(a).runZero
+  }
+
+  /** transforming every element using monadic transformation */
+  def transformM[A: Plated, M[_]: Monad](f: A => M[A])(a: A): M[A] = {
```

Comment:
It should be also an `Applicative` (for purpose of `modifyF`) so I can only reduce to `Bind with Applicative`. Do you think it will be better?